modules/channel-teams/src/backend/index.ts

```diff
@@ -0,0 +1,61 @@
+import 'bluebird-global'
+import * as sdk from 'botpress/sdk'
+import { Router } from 'express'
+
+import { Config } from '../config'
+
+import { setupMiddleware, TeamsClient } from './client'
+import { Clients } from './typings'
+
+const clients: Clients = {}
+
+const onServerReady = async (bp: typeof sdk) => {}
+
+const onServerStarted = async (bp: typeof sdk) => {
+  await setupMiddleware(bp, clients)
+}
+
+const onBotMount = async (bp: typeof sdk, botId: string) => {
+  const router = bp.http.createRouterForBot('channel-teams', {
+    checkAuthentication: false
+  })
+  const config = (await bp.config.getModuleConfigForBot('channel-teams', botId)) as Config
+
+  if (config.enabled) {
+    const bot = new TeamsClient(bp, botId, config, router)
+    await bot.initialize()
+
+    clients[botId] = bot
+  }
```

Comment:
Creating router in the onBotMount means that only the latest bot will have its api/messages endpoint available. Slack has a good example of handling callbacks for different bot ids. (the router should be in onServerStarted, then passed down to individual bot mounts)

Reply:
good catch